components/TARGET_PSA/TARGET_TFM/COMPONENT_SPE/secure_fw/core/ipc/tfm_spm.c

```diff
@@ -481,25 +468,37 @@ int32_t tfm_memory_check(void *buffer, size_t len, int32_t ns_caller,
     }
 
     if (access == TFM_MEMORY_ACCESS_RW) {
-        err = tfm_core_has_write_access_to_region(buffer, len, ns_caller);
+        err = tfm_core_has_write_access_to_region(buffer, len, ns_caller,
+                                                  privileged);
     } else {
-        err = tfm_core_has_read_access_to_region(buffer, len, ns_caller);
+        err = tfm_core_has_read_access_to_region(buffer, len, ns_caller,
+                                                 privileged);
     }
-    if (err == 1) {
+    if (err == TFM_SUCCESS) {
         return IPC_SUCCESS;
     }
 
     return IPC_ERROR_MEMORY_CHECK;
 }
 
+uint32_t tfm_spm_partition_get_privileged_mode(uint32_t partition_idx)
```

Comment:
Personally I'd prefer something along the lines of '..._is_priviledged_mode()' that returns TRUE or FALSE, but that might be too many years of Java. The function as it stands will still need its (typeless) return value checking against one of these constants; an opportunity for a bug later...